x-pack/agent/pkg/agent/application/api_key_decorator.go

```diff
@@ -0,0 +1,76 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package application
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/elastic/beats/x-pack/agent/pkg/agent/errors"
+	"github.com/elastic/beats/x-pack/agent/pkg/agent/program"
+	"github.com/elastic/beats/x-pack/agent/pkg/agent/transpiler"
+)
+
+const (
+	esOutputKey       = "output.elasticsearch"
+	esOutputAPIKeyKey = "output.elasticsearch.api_key"
+	apiKeyKey         = "api_key"
+)
+
+func injectESOutputAPIKey(
+	APIKey string,
+) (decoratorFunc, error) {
+
+	APIStrdec, err := base64.StdEncoding.DecodeString(APIKey)
+	if err != nil {
+		return nil, errors.New(errors.TypeConfig, "fail to decode API key")
```

Comment:
can you inject err as well? it will behave as traditional `errors.Wrap(err, "fail to ....")`